pkg/helm/helm.go

```diff
@@ -248,3 +259,34 @@ func (r *Impl) ChartPull(ref string) error {
 	var buf bytes.Buffer
 	return push.Run(&buf, ref)
 }
+
+// GetResources returns allocated resources for the specified release (their current state)
+func (r *Impl) GetResources(kubeNamespace string, releaseName string) ([]*unstructured.Unstructured, error) {
+	resources := make([]*unstructured.Unstructured, 0)
+	var rel *release.Release
+	var config *action.Configuration
+	var err error
+	var resourceList kube.ResourceList
+	config, err = getConfig(kubeNamespace)
+	if err != nil || config == nil {
+		return resources, err
+	}
+	status := action.NewStatus(config)
+	rel, err = status.Run(releaseName)
+	if err != nil {
+		return resources, err
+	}
+	resourceList, _ = config.KubeClient.Build(bytes.NewBufferString(rel.Manifest), false)
+	for _, res := range resourceList {
+		if err := res.Get(); err != nil {
+			return resources, err
+		}
+		obj := res.Object
+		if unstr, ok := obj.(*unstructured.Unstructured); ok {
```

Comment:
Why are we sure that this is always convertible?

Reply:
runtime object is always convertible. In any case we check for an error

Reply:
We report an error if it is an invalid run-time object, if it is not.

Reply:
I understand that, but getting an error is not desired here. My question is about the fact that the interface in `res` returns an Object and you assume it's always `unstructured.Unstructured`. It might be true, but that's an implementation detail that can change. Unless there is some magic allowing Object to be convertible to unstructured.Unstructured via plain cast and I'm not aware of it.

Reply:
I rely on the KubeClient.Build interface that returns unstructured objects:
`// Build validates for Kubernetes objects and returns unstructured infos.`